packages/core/src/service/plugin/index.js

```diff
@@ -162,6 +176,51 @@ export default class PluginService {
     ...this.plugins.layout.map(({ name }: Plugin) => name)
   ]
 
+  migratePluginState = (state: any, plugin: Plugin, dataVersion: string): ?Object => {
+    if (!plugin || !dataVersion || semver.valid(dataVersion) === null) {
+      return state
+    }
+    let currentDataVersion = dataVersion
+    let migrations = plugin.migrations ? plugin.migrations : []
+    while (true) {
+      const migration = migrations.find(m => semver.satisfies(currentDataVersion, m.fromVersionRange))
+      migrations = migrations.filter(m => !semver.satisfies(currentDataVersion, m.fromVersionRange))
+      if (!migration) {
+        // We assume all migrations necessary for the current version of plugin to work are provided
+        // Therefore if we don't find any, that means we are done and state is up to date
+        break
+      }
+      currentDataVersion = migration.toVersion
+      state = migration.migrate(state)
+    }
+    return state
+  }
+
+  getNewPluginState = (found: any, state: any, version: string) => {
+    if (!found.pluginWrongVersion || semver.lt(found.pluginWrongVersion.version, version)) {
```

Comment:
@PeterKottas if I read this correctly, the idea behind this is that, when calling `unserialize` with a specific plugin version, then `pluginWrongVersion` will be set if said plugin version does not exist in the `PluginService`, right? If so, a couple of questions:

1. Is the idea of `semver.lt(found.pluginWrongVersion.version, version))` here that, if we can not find right version, and the fallback plugin `*` is newer than the content version, we simply return the state with the `PluginMissing` plugin?
2. Does that imply that `pluginWrongVersion` is basically like an indicator that said plugin version was not found, but if a newer version was found we'll apply the migrations?
3. Assuming that we have two versions registered `0.0.2` and `0.0.3` and our content model is in `0.0.1`, would this work with this code? Seeing that you're returning the first best match to `*`?